pkg/apis/core/helper/qos/qos.go

```diff
@@ -113,11 +113,39 @@ func ComputePodQOS(pod *core.Pod) core.PodQOSClass {
 		allContainers = append(allContainers, pod.Spec.InitContainers...)
```

Comment:
@kannon92 I want to revert this back :|
Doesn't help much with the diff in the PR. Also, there's test coverage for this package. Let me know your thoughts

Reply:
My main concern is that we want to isolate this change to just your feature. If you sprinkle in refactors and cleanups it can cause problems.

I think your PR is an improvement but it is also mixed in with a pretty large feature.